task17/index.js

```diff
@@ -0,0 +1,115 @@
+  loadingPlace.id = "js-loadingPlace";
+  slideShowWrapper.appendChild(loadingPlace);
+};
+
+const removeLoading = () => document.getElementById("js-loadingPlace").remove();
+
+const renderErrorMessage = (errorMessage) => {
+  slideShowWrapper.textContent = errorMessage;
+};
+
+const getFetchData = async () => {
+  const response = await fetch(URL);
+  if (!response.ok) {
+    console.error(`${response.status}:${response.statusText}`);
+    renderErrorMessage("問題が発生し表示することができません。");
+  }
+  const json = await response.json();
+  return json;
+};
+
+const fetchImgData = () => {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(getFetchData(URL));
+    }, 3000);
+  });
+};
+
+const init = async () => {
+  addLoading();
+  try {
+    const value = await fetchImgData();
+    renderSlideImage(value);
+    renderSlideNumber(value);
+    addEventListenerForNextBtn(value);
+    addEventListenerForPrevBtn(value);
+  } catch (error) {
+    console.error(error.message);
+    renderErrorMessage("表示することができません");
+  } finally {
+    removeLoading();
+  }
+};
+init();
+
+const renderSlideImage = (value) => {
+  const slideImgFragment = document.createDocumentFragment();
+  for (let i = 0; i < value.length; i++) {
+    const slideImgItem = createElementWithClassName("li", "slide-img-item");
+    const slideImage = createElementWithClassName("img", "slide-image");
+    slideImgItem.id = `js-slideImgItem${i}`;
+    slideImage.id = `slide-image${i}`;
+    slideImage.src = value[i].img;
+    slideImage.alt = value[i].alt;
+    value[i].display && slideImgItem.classList.add("is-active");
+    slideImgFragment.appendChild(slideImgItem).appendChild(slideImage);
+  }
+  slideShowWrapper.appendChild(slideImgList).appendChild(slideImgFragment);
+};
+
+const renderSlideNumber = (value) => {
+  const slideNumText = createElementWithClassName("p", "slide-number");
+  const slideLength = value.length;
+  slideNumText.id = "js-slideNumber";
+  slideNumText.textContent = `1 / ${slideLength}`;
+  slideShowContainer.appendChild(slideNumText);
+};
+
+const nextBtn = document.getElementById("js-nextBtn");
+const prevBtn = document.getElementById("js-prevBtn");
+let imgNum = 0;
+
+const addEventListenerForNextBtn = (value) => {
+  nextBtn.addEventListener("click", () => {
+    document.querySelector(".is-active").classList.remove("is-active");
+    imgNum += 1;
+    document.getElementsByClassName("slide-img-item")[imgNum].classList.add("is-active");
+    btnDisabled(value);
+    renderActiveNumber(value);
+  });
+};
+
+const addEventListenerForPrevBtn = (value) => {
+  prevBtn.disabled = imgNum === 0;
+  prevBtn.addEventListener("click", () => {
+    document.querySelector(".is-active").classList.remove("is-active");
+    imgNum -= 1;
+    document.getElementsByClassName("slide-img-item")[imgNum].classList.add("is-active");
+    btnDisabled(value);
+    renderActiveNumber(value);
+  });
+};
```

Comment:
IMO
対象箇所を以下のようにまとめることも可能です。
(変数名は仮ですので、適切に修正をお願いします。)

buttonに共通のclassを付与し、イベントを定義しました！
const btn = document.querySelectorAll(".btn");
btn.forEach((b) => {
  b.addEventListener("click", (event) => {
    const target = event.target.parentElement;
    const img = document.querySelectorAll(".slide-img-item");
    document.querySelector(".is-active").classList.remove("is-active");
    target.id === "js-nextBtn" ? ++imgNum : --imgNum;
    img.classList.add("is-active");
    btnDisabled(img);
    renderActiveNumber(img);
  });
});
```

Reply:
ご教授ありがとうございます！凄過ぎて言葉もありません、、、！！
なぜ動くのか理解できていないので少しづつ理解したいです。
4e38e3b

Reply:
Sorry for omitting the explanation.
If you have any questions, please let me know!
It is better to write code with proper understanding so that we can support future enhancements.